crates/ty_python_semantic/src/semantic_index/builder.rs

```diff
@@ -2118,19 +2123,20 @@ impl<'ast> Visitor<'ast> for SemanticIndexBuilder<'_, 'ast> {
             ast::Expr::Name(ast::ExprName { ctx, .. })
             | ast::Expr::Attribute(ast::ExprAttribute { ctx, .. })
             | ast::Expr::Subscript(ast::ExprSubscript { ctx, .. }) => {
-                if let Ok(place_expr) = PlaceExpr::try_from(expr) {
-                    let mut place_expr = PlaceExprWithFlags::new(place_expr);
-                    if self.is_method_of_class().is_some()
-                        && place_expr.is_instance_attribute_candidate()
-                    {
-                        // We specifically mark attribute assignments to the first parameter of a method,
-                        // i.e. typically `self` or `cls`.
-                        let accessed_object_refers_to_first_parameter = self
-                            .current_first_parameter_name
-                            .is_some_and(|fst| place_expr.expr.root_name() == fst);
-
-                        if accessed_object_refers_to_first_parameter && place_expr.is_member() {
-                            place_expr.mark_instance_attribute();
+                if let Some(mut place_expr) = PlaceExpr::try_from_expr(expr) {
+                    if self.is_method_of_class().is_some() {
+                        if let PlaceExpr::Member(member) = &mut place_expr {
+                            if member.is_instance_attribute_candidate() {
```

Comment:
nit: it might be useful to reduce the nesting here by chaining the calls

Reply:
I think that requires let chains.